### PR TITLE
ScaledImage: don't scale img with native res

### PR DESCRIPTION
**Pull request by Apaczer**
what anoyed me from the beginning is that the `pv2x` tries to downscale the img even with 1:1 size (same as screen).

Let's make scaling conditional, and not set to true (`isScaled=1`) all the time.

### CODE REVIEW

**Comment by Apaczer** [comment]
on a second thought let's make it an OR

**Comment by szymor** [comment]
What if we have an image of 320x100000? What if 10000000x240? Have you tested such cases?

**Comment by Apaczer** [comment]
yes tks! Let's make it an AND also !

File: scaledimage.cpp [hunk 1]
```diff
@@ -13,6 +14,11 @@ ScaledImage::ScaledImage(std::string filename, int rotateMode) {
 		return;
 	}
 	
+	if (tmpload->w==SCREEN_WIDTH && tmpload->h<=SCREEN_HEIGHT || tmpload->h==SCREEN_HEIGHT && tmpload->w<=SCREEN_WIDTH)
```
Comment (szymor) [inline scaledimage.cpp]: The condition seems superfluous to me, isn't the second part enough?

File: scaledimage.cpp [hunk 2]
```diff
@@ -84,7 +90,6 @@ void ScaledImage::scale(double factor, double angle) {
 	}
 	
 	this->scaledImage=sdlCreateSurface((int)(this->image->w*65536.0*factor)>>16, (int)(this->image->h*65536.0*factor)>>16,1);
-	this->isScaled=1;
```
Comment (szymor) [inline scaledimage.cpp:87]: Are you sure this change actually does what you expect? To me it seems isScaled is just a flag signalling usage of scaledImage member. I have not delved much into the code, though. Just please double check you know what you are doing.

**Review by szymor — CHANGES_REQUESTED** [review]
Please review my comments.

File: scaledimage.cpp (hunk 1 above)
Comment (Apaczer) [reply scaledimage.cpp]: indeed, removed 2nd part but left if statement for readability

File: scaledimage.cpp (hunk 2 above)
Comment (Apaczer) [reply scaledimage.cpp:87]: sure I try to do a test run with every change. See `blit` member, if we dont remove that statement u will always have "imageScaled" blitted to Surface

Comment (szymor) [reply scaledimage.cpp:87]: For me it looks like a memory leak scenario. If you call `ScaledImage::scale` twice without `isScaled` flag set, you end up with an unallocated `scaledImage` surface. I would rather not call `scale` at all, if I was not interested in a scaled image. `scale` could possibly be called dependent on `isScaled` flag, but set in `init` method - it is just an idea which also needs to be reviewed before actual implementation.

Run valgrind to be 100% sure.

Comment (Apaczer) [reply scaledimage.cpp:87]: The flag is set in `init` member and if that's true then `isScaled` was badly written in the first place, which has been already proven. However not planning to do a bigger redraw ATM.

Probably some return func would suffice to not call scale twice, but haven't figure out it.

Comment (szymor) [reply scaledimage.cpp:87]: Unfortunately, it is your changes that could be problematic in terms of memory management here, so I cannot merge them until the issue is solved.

File: scaledimage.cpp [hunk 3]
```diff
@@ -85,57 +85,57 @@ void ScaledImage::scale(double factor, double angle) {
 	if (this->isScaled) {
 		SDL_FreeSurface(this->scaledImage);
 		this->scaledImage=NULL;
```
Comment (szymor) [inline scaledimage.cpp:87]: This line is superfluous.

**Review by szymor — COMMENTED** [review]
I am not so sure about actual effect of these changes on program execution as I did not run this code, but I cannot see any trivial mistakes this time.

Review my comments and let me know what you think about them.

File: scaledimage.cpp (hunk 3 above)
Comment (Apaczer) [reply scaledimage.cpp:87]: May be rm after merging, did not introduce that assign.
@szymor I'm working on a release on my dowstream fork, hope u dont mind

Comment (szymor) [reply scaledimage.cpp:87]: I know you did not, but you added another line below, which makes this line not needed anymore.

Feel free to work on your fork, ping me if you finish.

Comment (Apaczer) [reply scaledimage.cpp:87]: seems I missed the whole point of isScaled functionality in `scale` member, I'll make another approach

**Pull request closed without merging**